### PR TITLE
remove copystructure Mutex rewrite

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -853,22 +851,8 @@ func (ep *IstioEndpoint) DeepCopy() *IstioEndpoint {
...
 }
 
-// Configure copystructure so that it will not copy mutexes.
-var copyInternalConfig = copystructure.Config{
-	Copiers: map[reflect.Type]copystructure.CopierFunc{
-		reflect.TypeOf(sync.Mutex{}): func(interface{}) (interface{}, error) {
-			// Return a new mutex.
-			return sync.Mutex{}, nil
-		},
-		reflect.TypeOf(sync.RWMutex{}): func(interface{}) (interface{}, error) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/service.go:863]: This is not needed, ClusterVIPs deepcopy is not using copystructure

**Comment by hzxuzhonghu** [comment]
cc @ramaraochavali

**Review by howardjohn — APPROVED** [review]

**Pull request merged**